### PR TITLE
feat(behavior_path_planner): update resampling function

### PR DESCRIPTION
**Pull request by purewater0901**
Signed-off-by: yutaka <purewater0901@gmail.com>

## Description
Use resampling functions from motion utils and also resample input points. Currently, resample functions in the behavior path planner do not resample input path points, and due to this problem, it sometimes samples points without lanes in the input path when the sampling interval is large. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by tkimura4** [comment]
I confirmed that the bug that behavior modules such as intersection chatter was fixed by this PR.

Before:
https://user-images.githubusercontent.com/59680180/182775101-aff2450b-c1bf-4ff1-aba7-f4b3b38fb223.mp4

After:
https://user-images.githubusercontent.com/59680180/182774926-8b102cb5-a9b3-4319-89d9-e0fd2ea6a422.mp4

File: planning/behavior_path_planner/src/path_utilities.cpp [hunk 1]
```diff
@@ -89,80 +90,40 @@ double calcPathArcLength(const PathWithLaneId & path, size_t start, size_t end)
...
-
-  // For Point X, Y, Z
-  for (size_t i = 0; i < sampling_points.size(); ++i) {
-    PathPointWithLaneId p{};
-    p.point.pose.position =
-      tier4_autoware_utils::createPoint(resampled_x.at(i), resampled_y.at(i), resampled_z.at(i));
-    resampled_path.points.push_back(p);
-  }
+  constexpr double epsilon = 0.01;
+  const auto has_almost_same_value = [&](const auto & vec, const auto x) {
```
Comment (taikitanaka3) [inline planning/behavior_path_planner/src/path_utilities.cpp:108]: @purewater0901 
[Q] how was almost_same_value(point) handled in behavior path planner previously?

Comment (purewater0901) [reply planning/behavior_path_planner/src/path_utilities.cpp:108]: They previously do not sample input points, so it does not handle duplicated points

**Comment by purewater0901** [comment]
Since the number of failures are increasing when including this PR, please do not approve this PR until we find the problem

**Review by tkimura4 — APPROVED** [review]

**Pull request merged**